presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -78,4 +80,17 @@ public QueryStateInfoResource(
                         queryManager.getQueryResourceGroup(queryInfo.getQueryId()).map(ResourceGroupId::getRoot).map(rootResourceGroupInfos::get)))
                 .collect(toImmutableList());
     }
+
+    @GET
+    @Path("queryId/{queryId}")
```

Comment:
Ya, we talked about the other one last time and I suggested to use query param because when we initially discussed about this we were saying we may want to allow multiple usernames. I don't know whether we still want to do that. If not, the other one could be a path param as well. Though there is a bit of overhead to change the client side to hit different api but it's not too bad. Another option is to have both as query param.